# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/HostWebGLContext.cpp
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f dom/canvas/HostWebGLContext.cpp
--- a/dom/canvas/HostWebGLContext.cpp	Wed Jun 10 21:57:19 2020 +0000
+++ b/dom/canvas/HostWebGLContext.cpp	Wed Jun 10 22:21:02 2020 +0000
@@ -7,7 +7,7 @@
 
 #include "CompositableHost.h"
 #include "mozilla/layers/LayerTransactionChild.h"
-#include "mozilla/layers/TextureClientSharedSurface.h"
+#include "mozilla/layers/LayersSurfaces.h"
 
 #include "MozFramebuffer.h"
 #include "TexUnpackBlob.h"
@@ -106,9 +106,9 @@
   (void)mOwnerData.outOfProcess->mParent.SendJsWarning(text);
 }
 
-RefPtr<layers::SharedSurfaceTextureClient> HostWebGLContext::GetVRFrame(
-    ObjectId id) const {
-  return mContext->GetVRFrame(AutoResolve(id));
+Maybe<layers::SurfaceDescriptor> HostWebGLContext::GetFrontBuffer(
+    const layers::TextureType t) const {
+  return mContext->GetFrontBuffer(t);
 }
 
 //////////////////////////////////////////////